sidekick_core/lib/src/commands/update_command.dart

```diff
@@ -0,0 +1,69 @@
+import 'package:sidekick_core/sidekick_core.dart';
+import 'package:sidekick_core/src/version_checker.dart';
+
+/// Updates the sidekick cli
+class UpdateCommand extends Command {
+  @override
+  final String description = 'Updates the sidekick cli';
+
+  @override
+  final String name = 'update';
+
+  @override
+  Future<void> run() async {
+    final versionChecker = VersionChecker(Repository.requiredSidekickPackage);
+    // to remember which sidekick_core version the sidekick CLI was generated
+    // with, that sidekick_core version is written into the CLI's pubspec.yaml
+    // at the path ['sidekick', 'cli_version']
+
+    final latestSidekickCoreVersion =
+        await versionChecker.getLatestDependencyVersion('sidekick_core');
+    final currentSidekickCliVersion =
+        versionChecker.getMinimumVersionConstraint(['sidekick', 'cli_version']);
```

Comment:
Crashing for CLI's that do not contain `sidekic.cli_version` in `pubspec.yaml`. Meaning for all right now

Reply:
Fixed it and added a test for it

Reply:
Works :)